github/github.go

```diff
@@ -801,7 +801,11 @@ func parseTokenExpiration(r *http.Response) Timestamp {
 type requestContext uint8
 
 const (
-	bypassRateLimitCheck requestContext = iota
+	// BypassRateLimitCheck prevents a pre-emptive check for exceeded primary rate limits
+	// Specify this by providing a context with this key, e.g.
+	//   context.WithValue(context.Background(), github.BypassRateLimitCheck, true)
+	BypassRateLimitCheck requestContext = iota
```

Comment:
Why is `BypassRateLimitCheck` exported? Can't it remain `bypassRateLimitCheck`?

Reply:
Oh I missed that you wanted to trigger this from outside the library.
I suggest that there will be wrapper functions like:

```golang
func ContextWithBypassRateLimitCheck(ctx context.Context, bypass bool) context.Context {
    context.WithValue(ctx, BypassRateLimitCheck, bypass)
}
```

You don't even need to export the getter function for this value.

Anyway, as the code is now is fine